packages/dashboard/app/withdraw/utils/hiddenTokens.tsx

```
@@ -0,0 +1,117 @@
+            try {
+                const storedHiddenTokens = localStorage.getItem(HIDDEN_TOKENS_KEY);
+                if (storedHiddenTokens) {
+                    let parsedData;
+                    
+                    try {
+                        parsedData = JSON.parse(storedHiddenTokens);
+                    } catch (parseError) {
+                        console.error('Error parsing hidden tokens data', parseError);
+                        throw new Error('Invalid hidden tokens format');
+                    }
+                    
+                    try {
+                        const validatedData = hiddenTokensSchema.parse(parsedData);
+                        setHiddenTokens(validatedData);
+                    } catch (validationError) {
+                        console.warn('Hidden tokens data validation failed:', validationError);
+                        
+                        localStorage.removeItem(HIDDEN_TOKENS_KEY);
+                        setHiddenTokens([]);
+                        
+                        throw new Error('Hidden tokens data is invalid');
+                    }
+                }
+            } catch (error) {
+                console.error('Error loading hidden tokens from localStorage', error);
+                setError(error as string);
+                setHiddenTokens([]);
+                localStorage.setItem(HIDDEN_TOKENS_KEY, JSON.stringify([]));
+            } finally {
+                setIsProcessing(false);
+            }
+        }
+        loadHiddenTokens();
+    }, []);
+
+    const hideToken = useCallback((address: string) => {
+        if (address === 'ETH') return; // Don't allow hiding ETH
+
+        const lowercaseAddress = address.toLowerCase();
+        try {
+            ethereumAddressSchema.parse(lowercaseAddress);
+        } catch (validationError) {
+            console.error('Invalid address format:', validationError);
+            return;
```

Don't swallow and hide the error from the user